CDP4ServicesDal/CdpServicesDal.cs

```diff
@@ -247,6 +251,96 @@ public override async Task<IEnumerable<Thing>> Read(CDP4Common.DTO.Iteration ite
             return result;
         }
 
+        /// <summary>
+        /// Reads a physical file from a DataStore
+        /// </summary>
+        /// <param name="localFile">Download a localfile</param>
+        /// <param name="cancellationToken">
+        /// The <see cref="CancellationToken"/>
+        /// </param>
+        /// <returns>an await-able <see cref="Task"/> that returns a <see cref="byte"/> array.</returns>
+        public override async Task<byte[]> ReadFile(Thing localFile, CancellationToken cancellationToken) 

```

Comment:
would it be possible to type the `Thing localFile` as `FileRevision localfile` ?

Reply:
It would, but I would not recommend that.
I did a more generic to the "file upload" mechanisme.
That's also the reason that ILocalFile exists, for example. 
I could also have used FileRevision instead of ILocalFile.
I changed the parameter name to `thing` instead of `localFile` now and I propose to leave it as it is.